mlir/lib/IR/BuiltinAttributes.cpp

```diff
@@ -1822,3 +1823,43 @@ AffineMap mlir::makeStridedLinearLayoutMap(ArrayRef<int64_t> strides,
 
   return AffineMap::get(strides.size(), nSymbols, expr);
 }
+
+void mlir::printShape(OpAsmPrinter &printer, Operation *op,
+                      ArrayRef<int64_t> shape) {
+  if (!shape.empty())
+    printer << "[";
+
+  for (size_t i = 0; i < shape.size(); ++i) {
+    if (ShapedType::isDynamic(shape[i]))
+      printer << '?';
+    else
+      printer << shape[i];
+    if (i != shape.size() - 1) {
+      printer << 'x';
+    }
+  }
```

Comment:
We don't have a helper for this?

Reply:
[Here](https://github.com/llvm/llvm-project/blob/8eb705321ed20232aa13e85e07f22b44f19e82b4/mlir/lib/IR/AsmPrinter.cpp#L2577-L2593) is the implementation in the `AsmPrinter`. I will refactor out the common code into a new function.

Reply:
I added the new common function.